src/machines/should-i-train-machine.tsx

```diff
@@ -0,0 +1,161 @@
+import {assign, createMachine} from 'xstate'
+import {
+  Question,
+  getNextQuestion,
+  getPrevQuestion,
+} from 'utils/generate-question-states'
+import quiz from 'data/should-i-train.json'
+import {find} from 'lodash'
+
+export interface ShouldITrainQuizMachineContext {
+  responses: Score[]
+  currentQuestion: Question
+}
+
+interface Score {
+  score: number
+  selectedResponse: string
+  question: string
+  questionId: string
+}
+
+export type ShouldITrainQuizMachineEvent =
+  | {
+      type: 'BACK'
+    }
+  | {
+      type: 'CONFIRM'
+      data: Score
+    }
+
+const isScore = (eventData: any): eventData is Score => {
+  return (
+    (eventData as Score).question !== undefined &&
+    (eventData as Score).selectedResponse !== undefined &&
+    (eventData as Score).score !== undefined
+  )
+}
+
+const shouldITrainQuizMachine = createMachine<
+  ShouldITrainQuizMachineContext,
+  ShouldITrainQuizMachineEvent
+>(
+  {
+    id: 'shouldITrainForm',
+    initial: 'enteringTrained',
+    context: {responses: [], currentQuestion: quiz.questions[0]},
+    states: {
+      enteringTrained: {
+        on: {
+          CONFIRM: {
+            target: 'enteringSoreness',
+            actions: ['assignScore', 'assignCurrentQuestion'],
```

Comment:
we are doing these two actions on each event. This is where we will save the response to firebase as well.